internal/pkg/deploy/cloudformation/stack/transformers_test.go

```diff
@@ -1514,16 +1514,52 @@ func Test_convertPlatform(t *testing.T) {
 				},
 			},
 			out: template.RuntimePlatformOpts{
-				OS:   template.OSWindowsServerFull,
+				OS:   template.OSWindowsServer2019Full,
 				Arch: template.ArchX86,
 			},
 		},
-		"should return windows server core and x86_64 when platform is 'windows/x86_64'": {
+		"should return windows server 2019 core and x86_64 when advanced config specifies core": {
+			in: manifest.PlatformArgsOrString{
+				PlatformArgs: manifest.PlatformArgs{
+					OSFamily: aws.String(manifest.OSWindowsServer2019Core),
+					Arch:     aws.String(manifest.ArchX86),
+				},
+			},
+			out: template.RuntimePlatformOpts{
+				OS:   template.OSWindowsServer2019Core,
+				Arch: template.ArchX86,
+			},
+		},
+		"should return windows server 2022 full and x86_64 when advanced config specifies full": {
+			in: manifest.PlatformArgsOrString{
+				PlatformArgs: manifest.PlatformArgs{
+					OSFamily: aws.String(manifest.OSWindowsServer2022Full),
+					Arch:     aws.String(manifest.ArchX86),
+				},
+			},
+			out: template.RuntimePlatformOpts{
+				OS:   template.OSWindowsServer2022Full,
+				Arch: template.ArchX86,
+			},
+		},
+		"should return windows server 2022 core and x86_64 when advanced config specifies core": {
```

Comment:
Can we add one more unit test here for 2019 full? 🥺

Reply:
I do have all 4 variants for 2019 Full/2019 Core/2022 Full/2022 Core. Did I miss something? :)

Reply:
My bad! 🤦🏼